src/Collection/CollectionTrait.php

```
@@ -1195,23 +1195,29 @@ public function implode(string $glue, callable|string|null $path = null): string
     /**
      * @inheritDoc
      */
-    public function when(mixed $condition, callable $callback): CollectionInterface
+    public function when(mixed $condition, callable $callback, ?callable $else = null): CollectionInterface
     {
         if ($condition) {
             return $callback($this, $condition);
         }
+        if ($else) {
+            return $else($this);
+        }
 
         return $this;
     }
 
     /**
      * @inheritDoc
      */
-    public function unless(mixed $condition, callable $callback): CollectionInterface
+    public function unless(mixed $condition, callable $callback, ?callable $else = null): CollectionInterface
```

So `unless()` is now just `when()` with 2nd and 3rd arguments swapped. We can avoid increasing the API surface by only keeping `when()` will 2nd and 3rd arguments both nullable (we can throw an exception if  at least one of them is not passed).

---

so if the user wants a unless behavior they should use this? 
``` 
$collection->when($user->is_admin, else: fn($it) => $it->filter(....))
```
I'm not sure if this is clear

---

Why do unless() and when() need support for `else`? The two methods already give you a way to do if/else style filtering logic.

---

when is normally a conditional 'flow' with two parts

---

If your condition is the result of a function you don't need to assign the value to variable,

```php
//One method call
$collection->when(
    $Table->exists(['id' => 10]), 
    fn($it) => $it->filter(....), 
    fn($it) => $it->filter(....)
);

//instead of multiple calls
$collection->when(
    $Table->exists(['id' => 10]), 
    fn($it) => $it->filter(....),
)->unless(
    $Table->exists(['id' => 10]), 
    fn($it) => $it->filter(....),
);

//instead of using a var
$condition = $Table->exists(['id' => 10]);
$collection->when(
    $condition, 
    fn($it) => $it->filter(....),
)->unless(
    $condition,
    fn($it) => $it->filter(....),
);
```
```

---

To be fair I would prefer to negate the condition and have 
`$collection->when(!$user->is_admin, fn($it) => $it->filter(....))` than `$collection->when($user->is_admin, else: fn($it) => $it->filter(....))`